server/src/main/java/org/apache/druid/server/coordinator/DataSourceCompactionConfig.java

```diff
@@ -203,11 +197,34 @@ public int hashCode()
         keepSegmentGranularity,
         taskPriority,
         inputSegmentSizeBytes,
-        maxNumSegmentsToCompact,
         targetCompactionSizeBytes,
+        maxNumSegmentsToCompact,
         skipOffsetFromLatest,
         tuningConfig,
         taskContext
     );
   }
+
+  public static class UserCompactTuningConfig extends ClientCompactQueryTuningConfig
```

Comment:
why do we have two separate configs ? the only difference seems to be maxRowsPerSegment in these two.

Reply:
Good question. So, I added [maxRowsPerSegment as a constructor parameter](https://github.com/apache/incubator-druid/pull/6780/files#diff-97e8e8e2eb5a9f9ec96ab14f0fc467cdR67) because it's similar to `targetPartitionSize` and they should be at the same level. But, this can make users confused because `ClientCompactionQueryTuningConfig` also has `maxRowsPerSegment`. So I added `UserCompactTuningConfig` which doesn't have `maxRowsPerSegment`.